### ISSUE TITLE
Feature request: Nonzero exit status on failure.

### ISSUE THREAD

**Issue by FSMaxB** [TrustInSoft/tis-interpreter#91]
I'd like to see tis-interpreter exit with a nonzero exit status when an error occurs.

This would allow to include it into unit tests and continuous integration more easily (without having to look at the output in order to determine if the test failed.)

**Comment by pascal-cuoq** [comment]
I think I am going to make tis-interpreter exit with 1 if undefined behavior was found and 0 if no UB was found, even if the program itself called, say, `exit(2);`

The rationale is that I cannot imagine anyone using tis-interpreter because they want to learn a program's return code. Whereas I use tis-interpreter on testcases generated by fuzzing that make the tested program fail in most cases, and I am only interested whether there was UB, not in the return code.

**Comment by FSMaxB** [comment]
I agree. tis-interpreter should be used in addition to running regular tests. So regular test failures would be catched independently.

**Comment by FSMaxB** [comment]
But itw won't hurt to make this configurable via a command line flag, would it?

**Comment by pascal-cuoq** [comment]
It hurts in that it forces to write builtin implementations for `exit` and `abort`, which used to be ordinary non-terminating functions but become more than that if the return code is to be reported accurately.

**Comment by FSMaxB** [comment]
Hm. In that case just do what you think is the best solution. I personally don't require that option. Maybe others will have another opinion on that matter.

I know that valgrind has an option like that.

**Comment by regehr** [comment]
I like the idea of having the exit code reflect the definedness of the execution.

**Comment by pascal-cuoq** [comment]
Exit code 1 for static errors (this was already the case), 2 for dynamic errors, 0 for correct executions. I am sure that there are plenty of cases that are not covered yet, so keep grepping the logs for now, but it will get better progressively.

```
$ cat t.c ; tis-interpreter.sh t.c ; echo $?
int main(void) {
  return 09;
}
t.c:2:[kernel] failure: Invalid digit 9 in integer constant '09' in base 8.
[kernel] user error: stopping on file "t.c" that has errors. Add '-kernel-msg-key pp'
                     for preprocessing command.
[kernel] Frama-C aborted: invalid user input.
1
$ cat t.c ; tis-interpreter.sh t.c ; echo $?
int main(void) {
  return 90;
}
[value] Analyzing a complete application starting at main
[value] Computing initial state
[value] Initial state computed
[value] done for function main
0
$ cat t.c ; tis-interpreter.sh t.c ; echo $?
int main(void) {
  return 9/0;
}
[value] Analyzing a complete application starting at main
[value] Computing initial state
[value] Initial state computed
t.c:2:[kernel] warning: division by zero: assert 0 ≢ 0;
                  stack: main
[value] Stopping at nth alarm
2
```

**Comment by FSMaxB** [comment]
Good to hear. Thanks.

**Comment by o11c** [comment]
If used as a *wrapper* for unit tests, we still want it to be a failure if the unit tests fail.

There are plenty of cases of implementation-defined behavior, so it is quite likely that some tests will fail without UB just because it's a different environment, and that should be treated as a failure.
